dataflow/conftest.py

```diff
@@ -163,7 +163,12 @@ def pubsub_wait_for_messages(subscription_path: str) -> list[str]:
 
     subscriber = pubsub.SubscriberClient()
     with subscriber:
+        print(f"pulling from subscription_path")
         response = subscriber.pull(subscription=subscription_path, max_messages=10)
+
+        print("Raw data received:")
+        print([str(m.message.data) for m in response.received_messages])
```

Comment:
@jrmccluskey based on this debugging, your messages are coming back base64 encoded. 

From the debugging out of the last execution: 

```
Raw data received:
['b\'"zn\\xb58b\\xb2+\\x1a\\xb5\\xeb-~\\x8a\\xda?+a\\xa2{\\x1a\\x9a\\x99^:\\xebi\\xba\\xd9\\xe7Z\\xb8\\xady\\xa3\\xf2\\xb6\\x1a\\\'\\xa6\\xba \\xad\\xa9\\xad\\xa2)\\xa9\\x95\\xe9\\x9e\\x9e\\xd6\\x9b\\x8av\\xab\\xca\\xc7\\x9a\\xad\\xc8Z\\x96\\n+\\x8a\\xd8f\\xb6\\x87\\xe2\\x9d\\xdbaz\\x9a,\\x8a\\xd8\\xa8\\x9e\\x87\\xda\\xb2\\x97\\x9c\\x89\\xf8\\x9eu\\xe9^\\x99\\xe9\\xed\\x8av\\xa5\\x8a\\xcbg\\x9e\\x9c\\xad\\x86\\x89\\xe7u\\xe7\\xdb\\x8av\\xab\\xcb\\xfb\\x1ej\\xb7!\\x96+-\\xd6\\xd6\\xab\\x81\\xebg\\x96\\x8c\'']
```

With some slight manipulation to remove repr escapes (`s/\\/\`, `s/\'/'`), this noise looks like data. 

```
>>> data = b'"zn\xb58b\xb2+\x1a\xb5\xeb-~\x8a\xda?+a\xa2{\x1a\x9a\x99^:\xebi\xba\xd9\xe7Z\xb8\xady\xa3\xf2\xb6\x1a\"\xa6\xba \xad\xa9\xad\xa2)\xa9\x95\xe9\x9e\x9e\xd6\x9b\x8av\xab\xca\xc7\x9a\xad\xc8Z\x96\n+\x8a\xd8f\xb6\x87\xe2\x9d\xdbaz\x9a,\x8a\xd8\xa8\x9e\x87\xda\xb2\x97\x9c\x89\xf8\x9eu\xe9^\x99\xe9\xed\x8av\xa5\x8a\xcbg\x9e\x9c\xad\x86\x89\xe7u\xe7\xdb\x8av\xab\xcb\xfb\x1ej\xb7!\x96+-\xd6\xd6\xab\x81\xebg\x96\x8c"'
>>> 
>>> base64.b64encode(data)
b'InputThisisatestforaPythonsampleOutputnnWriteaPythoiprogramtoimplementabinarysearchalgorithmtofindthepositionofaspecifiedelementinalistnnpythonndefbinary/searchlist1targetnlowi'
```

My guess here there's a missing de/serialisation step somewhere.

Comparing with the existing gemma example, removing the differences with the implied `run` when using `with beam.Pipeline` ([source](https://stackoverflow.com/a/68718420/124019)), can you think of any other reason why the data would be half de/serialized like this? 

(Self note: The debugging in this particular part of the file needs to be removed before merging)

Reply:
This is strange. The write we do to Pub/Sub converts our string to bytes and writes to the pub sub topic. I don't think we're missing anything as far as getting the data out. The implicit run shouldn't matter here either